program-runtime/src/loaded_programs.rs

```diff
@@ -317,14 +321,44 @@ impl LoadedPrograms {
     }
 
     /// Evicts programs which were used infrequently
-    pub fn sort_and_evict(&mut self) {
-        // TODO: Sort programs by their usage_counter
-        // TODO: Truncate the end of the list
+    pub fn sort_and_evict(&mut self, max_cache_entries: Option<usize>) -> Vec<Pubkey> {
+        // Find eviction candidates and sort by their usage counters
+        let mut num_cache_entries = 0;
+        let sorted_candidates = self.entries
+            .iter_mut()
```

Comment:
Why does it need to be iter_mut()? The entries hashmap should not be modified by this operation.

Reply:
It doesn't, my bad.